docs/articles/documentation/test-api/authentication/user-roles.md

```diff
@@ -168,4 +176,27 @@ test('My test', async t => {
         .useRole(role);
 ```
 
-**Default value**: `false`
\ No newline at end of file
+**Default value**: `false`
+
+## Troubleshooting
+
+### Test Actions Fail After Authentication
+
+This issue can be caused by the browser's page caching.
+
+In order to guarantee seamless test execution, browsers that run TestCafe tests should always fetch the tested page from the TestCafe proxy server. This ensures that automation scripts on the page are in sync with the server side.
+
+However, if the browser uses a cached copy of the page, automation mechanisms may be interrupted. Among other issues, this could reset authentication data in the cookies, local and session storages during navigation.
```

Comment:
However, if the browser uses a cached copy of the page, automation mechanisms may be interrupted. Among other issues, this could reset authentication data in the cookies, local and session storages during navigation. [this last sentence doesn't make sense; reset data in cookies and storages?]

Reply:
> [this last sentence doesn't make sense; reset data in cookies and storages?]

[Cookies](https://en.wikipedia.org/wiki/HTTP_cookie) and [web storage](https://en.wikipedia.org/wiki/Web_storage) (the latter consists of local and session storage) are different ways to store data in the browser. In this phrase, I'm trying to say that data that has been saved in both places could be reset.

But "storage" is uncountable I guess, my mistake, so it should be:
*...this could reset authentication data in the cookies, local and session storage during navigation.*

Or it still doesn't make sense?

Reply:
yes, use that